### PR TITLE
Add SparsityPattern::write_svg

### PR DESCRIPTION
**Pull request by tcclevenger**

### CODE REVIEW

File: tests/lac/sparsity_pattern_write_svg_01.cc [hunk 1]
```diff
@@ -0,0 +1,38 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2008 - 2014 by the deal.II authors
```
Comment (bangerth) [inline tests/lac/sparsity_pattern_write_svg_01.cc]: Just list 2015 here.

File: include/deal.II/lac/sparsity_pattern.h [hunk 2]
```diff
@@ -928,6 +928,15 @@ class SparsityPattern : public Subscriptor
   void print_gnuplot (std::ostream &out) const;
 
   /**
+   * Prints the sparsity of the matrix in a .svg file which can be opened in a web browser.
+   * The .svg file contains pixels which correspond to the entries in the matrix. An entry
+   * in the matrix which contains a non-zero value corresponds with a red pixel while a
+   * zero-valued entry in the matrix correspond with a white pixel.
```
Comment (bangerth) [inline include/deal.II/lac/sparsity_pattern.h]: A pixel is really a physical representation of an element of your screen, or alternatively the smallest, indivisible piece of a picture. Pixels can't be scaled --  I think you want to call them "squares" in the documentation.

**Comment by bangerth** [comment]
Nice feature! Can you add an entry to `doc/news/changes.h` announcing the new feature?

File: source/lac/sparsity_pattern.cc [hunk 3]
```diff
@@ -856,6 +856,36 @@ SparsityPattern::print_gnuplot (std::ostream &out) const
...
+
+  SparsityPattern::iterator
+  it = this->begin(),
+  end = this->end();
+  for (; it!=end; it++)
+    {
+      out<<"  <rect class=\"pixel\" x=\""<<it->column()+1<<"\" y=\""<<it->row()+1<<
+         "\" width=\".9\" height=\".9\"/>\n";
+    }
+  out<<"</svg>";
```
Comment (bangerth) [inline source/lac/sparsity_pattern.cc]: Most file formats require that a file end in a newline. Can you add this here (and in the corresponding output file of the test)?

**Comment by bangerth** [comment]
Great, thanks. I've got only one more request that I forgot about last time. Can you add this, and then squash all of your patches into one?

Thanks!

**Comment by tcclevenger** [comment]
Done. Let me know if there are any problems.

**Comment by bangerth** [comment]
Excellent -- thanks for your contribution!

**Pull request merged**